core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -285,8 +287,20 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
         sqp.setDefaultOperator(defaultOperator.toBooleanClauseOccur());
 
         Query query = sqp.parse(queryText);
-        if (minimumShouldMatch != null && query instanceof BooleanQuery) {
-            query = Queries.applyMinimumShouldMatch((BooleanQuery) query, minimumShouldMatch);
+        if (query instanceof BooleanQuery) {
+            BooleanQuery booleanQuery = (BooleanQuery) query;
+            if (booleanQuery.clauses().size() > 1
+                    && ((booleanQuery.clauses().get(0)).getQuery() instanceof BooleanQuery) == false) {
```

Comment:
The problem in #13884 is that for one query term & multiple fields, `minimum_should_match` gets calculated based on the number of fields instead of just the one term like it should. As far as I can see this seems to be because SimpleQueryParser only starts combining the BooleanQueries that are produced for each term into new BooleanQueries for more than one term/subquery. 

What I'm trying to do here is applying this additional wrapping for the one-term case. The problem now is, how to distiguish the one-term/multi-field case from the multi-term/one-field case. Before this change, Lucenes SimpleQueryParser#simplify() performs some rewrite of one-clause subqueries, so the resulting queries for those two cases have the same structure:
```
field1: "T1 T2 T3" => (field1:T1 field1:T2 field1:T3) 
field1, field2, field3: "T1" => (field1:T1 field2:T1 field3:T1)
```
This PR removes this simplification, so now each term/subquery should be a BooleanQuery in itself:
```
field1: "T1 T2 T3" =>  ( (field1:T1)  (field1:T2) (field1:T3) )
field1, field2, field3: "T1" => (field1:T1 field2:T1 field3:T1)
```
Now I can detect the "problematic" case (one term, multiple fields), since it will have leaf TermQueries on the second level of the query tree. That's the case where we need one extra wrapping into an additinal BooleanQuery, so the `minimum_should_match` gets calculated correctly.